### ISSUE TITLE
Deleting entities within a service path deletes entities in sub-paths

### ISSUE THREAD

**Issue by LeandroGuillen** [telefonicaid/fiware-orion#885]
An issue has been raised at :
http://stackoverflow.com/questions/29761530/deleting-entity-in-a-fiware-servicepath-deletes-entities-with-the-same-id-in-su

It basically states that you will delete two entities with the same ID even if they are in different service paths, assuming one path is a sub-path of the other.

The way to replay this situation is as follows.

Create **table1** in service path **/user/home**:

```bash
(curl localhost:1026/v1/updateContext -s -S -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Fiware-ServicePath: /user/home' -d @- | python -mjson.tool) <<EOF
{
 "contextElements": [
   {
      "type": "table",
      "isPattern": "false",
      "id": "table1"
    }
  ],
  "updateAction": "APPEND"
}
EOF
```

Create **table1** in service path **/user/home/room**:
 
```bash
(curl localhost:1026/v1/updateContext -s -S -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Fiware-ServicePath: /user/home' -d @- | python -mjson.tool) <<EOF
{
 "contextElements": [
   {
      "type": "table",
      "isPattern": "false",
      "id": "table1"
    }
  ],
  "updateAction": "APPEND"
}
EOF
```

Right now, the situation in the database is:

```mongo
> db.entities.find()
{
    "_id": {
        "id": "table1",
        "type": "table",
        "servicePath": "/user/home"
    },
    "attrs": [],
    "creDate": 1429630324,
    "modDate": 1429630324
}
{
    "_id": {
        "id": "table1",
        "type": "table",
        "servicePath": "/user/home/room"
    },
    "attrs": [],
    "creDate": 1429630325,
    "modDate": 1429630325
}
```

Now we do a deletion on entity **table1** but indicating service path **/user**:

```bash
(curl localhost:1026/v1/updateContext -s -S -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Fiware-ServicePath: /user/home' -d @- | python -mjson.tool) <<EOF
{
 "contextElements": [
   {
      "type": "table",
      "isPattern": "false",
      "id": "table1"
    }
  ],
  "updateAction": "DELETE"
}
EOF
```

Now the database is empty:
```mongo
> db.entities.find()
> 
```

This is not an acceptable behaviour. What should happen is that only entities that reside in the specified service path header should get deleted.

Furthermore, if you do a deletion but on the root service path or ommit it:
```bash
(curl localhost:1026/v1/updateContext -s -S -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Fiware-ServicePath: /' -d @- | python -mjson.tool) <<EOF
{
 "contextElements": [
   {
      "type": "table",
      "isPattern": "false",
      "id": "table1"
    }
  ],
  "updateAction": "DELETE"
}
EOF
```
you get this:
```
{
	"contextResponses": [
	{
		"contextElement": {
			"type": "table",
			"isPattern": "false",
			"id": "table1"
		},
		"statusCode": {
			"code": "404",
			"reasonPhrase": "No context element found",
			"details": "table1"
		}
	}
	]
}
```

So this is another aspect of the same bug.

In short:
 1. Deleting entities with same id in different service paths should not be allow without explicitly using the # character.
 2. The behavior is not consistent in these two cases:
	- Parent service-path is root
	- Parent service-path is _not_ root

**Comment by fgalan** [comment]
Fixed in PR https://github.com/telefonicaid/fiware-orion/pull/924
